scenes/game_elements/characters/enemies/guard/components/guard.gd

```diff
@@ -32,6 +32,44 @@ const LOOK_AT_TURN_SPEED: float = 10.0
 			animated_sprite_2d.sprite_frames = sprite_frames
 		update_configuration_warnings()
 
+@export_category("Sounds")
+## Sound played when a guard's [enum State] enters DETECTING or ALERTED.
+@export var alerted_sound_stream: AudioStream = preload(
+	"res://assets/sound/third_party/enemies/guard/AlertSound.ogg"
+):
+	set(new_value):
+		alerted_sound_stream = new_value
+		if not is_node_ready():
+			await ready
+		_alert_sound.stream = new_value
```

Comment:
I find it a bit more readable when the setters are named functions, than when they are unnamed like in here.